### ISSUE TITLE
Cyanite queries very slow under moderate load

### ISSUE THREAD

**Issue by tehlers320** [pyr/cyanite#228]
I am getting really slow response times and im not sure if this is normal or not
```
 time curl localhost:8080/paths?query=collectd_metrics.cyanite-cassandra.ip-10-2-92-19.aggregation-cpu-average.percent-user
[{"text":"percent-user","id":"collectd_metrics.cyanite-cassandra.ip-10-2-92-19.aggregation-cpu-average.percent-user","path":"collectd_metrics.cyanite-cassandra.ip-10-2-92-19.aggregation-cpu-average.percent-user","allowChildren":false,"expandable":false,"leaf":true}]
real	0m10.484s

```

Queries direct to metrics are fine so this must be something with the paths feature

```
time curl 'localhost:8080/metrics?path=collectd_metrics.cyanite-cassandra.ip-10-5-5-1.disk-xvdb.disk_ops.write&from=1470829576&to=1470851176'
{"from":1470839100,"to":1470851100,"step":300,"series":{"collectd_metrics.cyanite-cassandra.ip-10-5-5-1.disk-xvdb.disk_ops.write":[0.583334,null,null,3.449995,18.250002,10.233332,26.523328,42.97667740000001,19.499997399999998,null,null,null,3.37916325,15.466662,3.883351,6.850008,46.3133462,null,null,null,null,14.903332,null,3.9416720000000005,7.3566682,25.290002999999995,54.56000300000001,25.589991599999998,50.933337400000006,46.8299864,38.71665,46.6033562,45.2899608,49.8733518,45.0799882,36.0100116,51.1867242,47.2366038,43.4833244,46.1366748,null]}}
real	0m0.630s

```

I don't think cassandra is the problem since all the latencys are very low on reads.

nodetool tablehistograms metric path

```
metric/path histograms
Percentile  SSTables     Write Latency      Read Latency    Partition Size        Cell Count
                              (micros)          (micros)           (bytes)                  
50%             0.00             29.52              0.00               770                 5
75%             0.00             35.43              0.00              1331                 7
95%             0.00             51.01              0.00              1597                12
98%             0.00             61.21              0.00              6866                35
99%             0.00             73.46              0.00              6866                35
Min             0.00              1.60              0.00               125                 0
Max             0.00          62479.63              0.00              6866                35

```
 nodetool tablehistograms metric segment

```
metric/segment histograms
Percentile  SSTables     Write Latency      Read Latency    Partition Size        Cell Count
                              (micros)          (micros)           (bytes)                  
50%             1.00             20.50            263.21            943127             14237
75%             2.00             24.60            943.13           4055269             51012
95%             3.00             42.51            943.13          17436917            182785
98%             3.00             51.01            943.13          25109160            315852
99%             3.00             61.21            943.13          25109160            315852
Min             1.00              0.00            152.32                51                 2
Max             4.00         107964.79            943.13          25109160            315852

```
nodetool tablehistograms metric metric

```
metric/metric histograms
Percentile  SSTables     Write Latency      Read Latency    Partition Size        Cell Count
                              (micros)          (micros)           (bytes)                  
50%             6.00             11.86            454.83               642                 6
75%             7.00             14.24            545.79              1597                20
95%             8.00             20.50            785.94              6866               103
98%            10.00             29.52           1131.75              8239               103
99%            10.00             29.52           1131.75             20501               310
Min             2.00              1.60             88.15               104                 0
Max            10.00         107964.79           1358.10             61214              1331`

```

I have a pretty large dataset sending here

```
select count(*) from metric.segment;
 count
 333752
(1 rows)
Warnings :
Aggregation query used without partition key

```

My setup is currently
cyanite on an m4.xlarge sending to 2x c3.4xlarge cassandra nodes (3.7)

Cyanite settings (compiled from source yesterday)

`java -server -ea -Xms8G -Xmx8G -Xloggc:/var/log/cyanite/gc.log cyanite/target/cyanite-0.5.1-standalone.jar -f /etc/cyanite.yaml
`

Config

```
logging:
  level: debug
  console: true
  files:
    - "/var/log/supervisor/cyanite-log-query.log"
input:
  - type: "carbon"
    host: 0.0.0.0
    port: 2009
store:
  cluster: 
    - "10.2.92.19"
    - "10.2.92.37"
  keyspace: "metric"
index:
  type: cassandra
  cluster:
    - "10.2.92.19"
    - "10.2.92.37"
  keyspace: "metric"
api:
  port: 8080
  host: 0.0.0.0
engine:
  rules:
    default:
      - "60s:1h"
      - "5m:90d"
      - "1h:365d"
```

**Comment by tehlers320** [comment]
The schema is copied right out of the docs.

**Comment by ifesdjeen** [comment]
@tehlers320 did you build from the master?...

**Comment by tehlers320** [comment]
Yes, i have 2 builds i've tried this was master. I just moved over to the fix-wildcard-queries branch and dropped the keyspace.

**Comment by ifesdjeen** [comment]
I think my latest commit to #216 should have improved / fixed the situation. For fully specified metrics we've been making a wildcard query with post-filtering. Sometimes on your dataset it may have resulted into fetching most of your dataset.

**Comment by tehlers320** [comment]
have let this bake for a day or 2 now under 333k metrics per min load. Verfied fixed with the #216 fix and its additional commit.

**Comment by ifesdjeen** [comment]
🎆
